MigrationTutorial/Models/V1/Consumable.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using MongoDB.Bson;
+using Realms;
+
+namespace MigrationTutorial.Models.V1
+{
+    public class Consumable : RealmObject
+    {
+        [PrimaryKey]
+        public ObjectId Id { get; private set; } = ObjectId.GenerateNewId();
+
+        public string ProductId { get; set; }
+
+        public ConsumableType Type
+        {
+            get => Enum.Parse<ConsumableType>(_Type);
+            set=> _Type = value.ToString();
+        }
+
+        [Required]
+        private string? _Type { get; set; } = string.Empty;
+
+        public int Quantity { get; set; } = 0;
+
+        [Required]
+        public string UnitOfMeasure {get; set;}
+
+        public float Price { get; set; }
+
+        private Consumable() { }
+
+        public Consumable(string productId = "")
+        {
+            ProductId = productId;
+        }
+    }
+
+
+    public enum ConsumableType
+    {
+        Glue = 1,
```

Comment:
I am not sure I understand. The way I store the value is by string, the name of the enum itself, and the getter translates from the string. What does it matter the baking value? Your point would be correct if those values change at runtime.
The problem that I believe you're referring to appears when you store an int instead of a string in realm. Then in that case, yes, if you modify the value of the enum (maybe reordering them or what not) then translating from int would give the wrong enum.

Reply:
I see! I like that approach. Make it safe against reordering. Remaking one of those would then have to be handled in a migration right?

Reply:
What do you mean by "remaking one of those"?